osquery/tables/ipmi/ipmi_info.cpp

```diff
@@ -0,0 +1,1297 @@
+          d->client->insertRowsQueue(
+              d->mcKey, d->colName, toMAC(parmVal, dataLen));
+          break;
+
+        case IPMI_LANPARM_IP_ADDRESS_SRC:
+          d->client->insertRowsQueue(
+              d->mcKey, d->colName, toAddressSrc(parmVal, dataLen));
+          break;
+
+        case IPMI_LANPARM_COMMUNITY_STRING: {
+          std::string comm(reinterpret_cast<const char*>(parmVal + 1),
+                           static_cast<size_t>(dataLen));
+          d->client->insertRowsQueue(d->mcKey, d->colName, comm);
+          break;
+        }
+
+        case IPMI_LANPARM_SET_IN_PROGRESS:
+          d->client->insertRowsQueue(
+              d->mcKey, d->colName, toSetInProgress(parmVal, dataLen));
+          break;
+
+        default:
+          LOG(ERROR) << "Got an unexpected LANPARM type: "
+                     << std::to_string(d->parm);
+          break;
+        }
+
+      },
+      data);
+
+  if (rv != 0) {
+    LOG(ERROR) << "Could not get default gateway lanparm: " << strerror(rv);
+  }
+}
+
+/*===========================OpenIPMI Callbacks================================
+ */
+
+void getLANsCB(ipmi_domain_t* domain, ipmi_mc_t* mc, void* data) {
+  IPMIClient* client = reinterpret_cast<IPMIClient*>(data);
+
+  std::string key = getMCKey(mc);
+  ipmi_lanparm_t* lp = client->getLANParm(key, mc);
+  if (lp == nullptr) {
+    LOG(ERROR) << "Could not get ipmi_lanparm_t for mc '" << key;
+    return;
+  }
+
+  char name[IPMI_MC_NAME_LEN];
+  int len = ipmi_mc_get_name(mc, name, IPMI_MC_NAME_LEN);
+  if (len > 0) {
+    client->insertRowsQueue(key, "mc_name", name);
+  }
+  client->insertRowsQueue(key, "mc_id", INTEGER(ipmi_mc_device_id(mc)));
+
+  getPARM(lp, IPMI_LANPARM_IP_ADDRESS, key, "ip", client);
+
+  getPARM(lp, IPMI_LANPARM_DEFAULT_GATEWAY_ADDR, key, "gateway_ip", client);
+
+  getPARM(lp, IPMI_LANPARM_MAC_ADDRESS, key, "mac", client);
+
+  getPARM(lp, IPMI_LANPARM_SUBNET_MASK, key, "subnet_mask", client);
+
+  getPARM(
+      lp, IPMI_LANPARM_DEFAULT_GATEWAY_MAC_ADDR, key, "gateway_mac", client);
+
+  getPARM(
+      lp, IPMI_LANPARM_BACKUP_GATEWAY_ADDR, key, "backup_gateway_ip", client);
+
+  getPARM(lp,
+          IPMI_LANPARM_BACKUP_GATEWAY_MAC_ADDR,
+          key,
+          "backup_gateway_mac",
+          client);
+
+  getPARM(lp, IPMI_LANPARM_IP_ADDRESS_SRC, key, "ip_address_source", client);
+
+  getPARM(
+      lp, IPMI_LANPARM_COMMUNITY_STRING, key, "snmp_community_string", client);
+
+  getPARM(lp, IPMI_LANPARM_SET_IN_PROGRESS, key, "set_in_progress", client);
+}
+
+void IPMIFullyUpCB(ipmi_domain_t* domain, void* data) {
+  TLOG << "OpenIPMI is now fully up";
+  IPMIClient* c = reinterpret_cast<IPMIClient*>(data);
+  c->setDomain(domain);
+
+  return;
+}
+
+/// Gets the the value suffix for a sensor reading.
+std::string getSensorThresholdSuffix(ipmi_sensor_t* sensor) {
+  std::string percent, base, mod_use, modifier, rate;
+
+  base = ipmi_sensor_get_base_unit_string(sensor);
```

Comment:
nit: you might also be able to get away with an `auto base = ipmi_sensor_get_base_unit_string(sensor);`